packages/@ember/-internals/routing/lib/utils.ts

```diff
@@ -8,20 +8,28 @@ import EmberRouter, { PrivateRouteInfo, QueryParam } from './system/router';
 
 const ALL_PERIODS_REGEX = /\./g;
 
-export function extractRouteArgs(args: any[]) {
+export function extractRouteArgs(args: unknown[]) {
   args = args.slice();
   let possibleQueryParams = args[args.length - 1];
 
   let queryParams;
-  if (possibleQueryParams && possibleQueryParams.hasOwnProperty('queryParams')) {
-    queryParams = args.pop().queryParams;
+  if (possibleQueryParams && (possibleQueryParams as object).hasOwnProperty('queryParams')) {
```

Comment:
We shouldn't use this object's `.hasOwnProperty` 😢 (this breaks when you pass in the result of `(hash` for example, because there is no prototype)

Reply:
Yeah, I assumed this was a bug for exactly that reason, but opted very carefully not to make any functional changes in this PR. I'm very happy to submit a follow-on for that, but wanted to keep this one to runtime-safe docs-and-types-only changes.

Reply:
Sounds good, mostly just reading the PR made me cringe a bit 🗡